indra/newview/llviewermessage.cpp

```diff
@@ -3228,7 +3197,22 @@ void send_agent_update(bool force_send, bool send_reliable)
                             AGENT_CONTROL_ML_LBUTTON_UP ;
     }
 
-    control_flag_change = last_control_flags ^ control_flags;
+    // any change in control_flags should be sent ASAP, so we fold that into force_send
+    force_send = force_send || (control_flags != last_control_flags);
```

Comment:
This is the main fix right here: when `control_flags` have changed we send an update ASAP regardless of send rate limits.  The rest of the changes are cleanup and reorganizing how all of the other conditions are checked.